### DISCUSSION TITLE
[Solved] Why Filament pages slow taking 10s to load initial html?

### DISCUSSION THREAD

**Discussion by christoferd** [filamentphp/filament#6120] [Help] [answered]
**What have I done wrong? Filament pages running extremely slow.**
Note that all the load times examples are 2nd attempts with server cache (apcu) and browser cache enabled (unless the image shows cache disabled).

#### TLDR; Update 5-Apr-2023
Filament Dashboard (Debug & Debug Bar = true):
Local PC & Staging servers: ~1.6s

Other dashboard (no Filament)
Local PC & Staging servers: ~0.8s
- Filament page takes 2x longer to load with just the base layout view including navigation. Why?

**Noting differences between Local PC and Remote Server**
* Remote runs FPM/FastCGI - local PC doesn't
* Remote runs Thread-Safe disabled - local PC runs Thread Safe
* Remote server CPU is probably much faster
* Remote server is not Windows

**Filament Table/List:**
10 records per page
3 columns, each cell is related information from another table.
Local PC (Debug & Debug Bar On): 
* 174 models!
* 41MB!
* 11.32s!
![image](https://user-images.githubusercontent.com/1835883/230112563-57bf4ffc-0df1-4cbd-b4e3-4b475f6f3549.png)

Local PC (Debug & Debug Bar Off): 
* 7s!

Staging Server (Debug & Debug Bar Off): 
* 1.3s
**Why is the remote server running faster? What tools are there to troubleshoot this?**
Note that changing APP_ENV=local to something else, has no effect on the load times.

Everything is setup following instructions from the Filament documentation site. I'm new to this, so I've only been following the instructions.

I would like to set this to 50 records per page, but then it will reach the 40+seconds load time 👎 
I'm guessing there's a better way to do this, by overriding the query used to grab the records, but it goes against the principle reason for using this Filament tool/framework. And then if I did try do it, I can see issues when I want to use ordering, filters etc.

## Staging Server 5-Apr-2023
This runs similar to local PC. Weird though, because debug bar is off, so I'd expect it to go much faster!
APP_DEBUG=true
DEBUGBAR_ENABLED=false

Filament Dashboard - blank, without Widgets
![image](https://user-images.githubusercontent.com/1835883/230108966-f1086930-3580-4dd4-891d-701cd1b012ba.png)

My dashboard - no Filament

![image](https://user-images.githubusercontent.com/1835883/230109261-57186e95-7436-410f-bc97-626692519709.png)

## Local Windows 11

This is a new app, Laravel v10.5.1
PHP 8.1.17
Windows 11
Server using CLI: php artisan serve running on 127.0.0.1:8000
Laravel cache "apc"
No memcached
No redis
Database local Maria 10.x

I have reached 1590 views in a single request! 48MB! So looking at other discussions, this may be something to do with DebugBar?... but Filament is definitely doing something to slow down the app.

![image](https://user-images.githubusercontent.com/1835883/229908912-bb114c82-d237-49e4-aab2-b7b633339de1.png)

Notice that it took 42s to load. That's on my development notebook. Recently purchased, running M.2 drive.
All other apps run very quick.

As I add more resources, the loading time increases. I don't understand why.

I can post some pages of content, but I can't just dump all my code for public view.

With APP_DEBUG=false & DEBUGBAR_ENABLED=false...

![image](https://user-images.githubusercontent.com/1835883/229910073-2e90d7fc-8974-45f4-8df5-96612cf83324.png)

This is one of my own routes that doesn't use filament. Notice the 390ms initial file load... now compare that to the previous one!

![image](https://user-images.githubusercontent.com/1835883/229910412-05f77913-84d4-482c-a66d-e25cdeb3670e.png)

## Composer
![image](https://user-images.githubusercontent.com/1835883/229917008-e5e00575-37d4-451b-8914-c73d5c8b2a2a.png)
![image](https://user-images.githubusercontent.com/1835883/229917116-e4b75684-8742-4372-bf9f-36d26a738e98.png)

## More troubleshooting...
No Debug, after setting google_fonts to NULL, it still takes the same amount of time, 10s to load first html.

![image](https://user-images.githubusercontent.com/1835883/229913155-58f3bac7-ffe1-452b-a5b6-f63f1765de37.png)

I enabled OpCache, it now loads first html in 9.3 seconds :(

## Switched to Apache 2.4
Using XAMPP instead now, same PHP v8.1.X
APP_DEBUG=false & DEBUGBAR_ENABLED=false
It takes 1.2 seconds to load initial html. That is still very slow compare to the other page within the same app.
![image](https://user-images.githubusercontent.com/1835883/229918208-c467947b-f74a-43f0-bd26-28790f2a6f4d.png)

## Debug back on
APP_DEBUG=true & DEBUGBAR_ENABLED=true
5.89s
![image](https://user-images.githubusercontent.com/1835883/229923780-738cbbe4-742b-4ba1-a8fc-678ea578f8b7.png)

vs my own dashboard route that doesn't use Filament...
0.69s
![image](https://user-images.githubusercontent.com/1835883/229924023-6164d199-31fe-4614-b29a-fd8b24914294.png)

**Queries**
What happens if I keep going and add another 20 resources?

Why is Filament acting so differently with artisan serve (PHP CLI server) vs Apache?

**Comment by danharrin** [comment]
In your service providers, do you have any code that shares data with all views?

The reason why there is a high view count is because we use Blade components, but simply rendering them should not take that long.

**Comment by christoferd** [comment]
Hi Dan, thanks for response. This app is fairly new. AppServiceProvider is bare.
PS: I'm updating original post with more information after publishing as a Staging test on Cloudways server 

![image](https://user-images.githubusercontent.com/1835883/230107419-694e1894-debc-4d0a-b982-bd787c1c54a4.png)

**Reply by danharrin** [reply]
What other packages and plugins have you installed

**Reply by pathros** [reply]
In my case, the installation and use of several [blade icons](https://blade-ui-kit.com) packages were causing the FilamentPHP to get extremely slow. As @relayer464 reported, the execution of `php artisan icons:cache` fixed the slowness problem.

**Reply by TilsonM17** [reply]
> In my case, the installation and use of several [blade icons](https://blade-ui-kit.com) packages were causing the FilamentPHP to get extremely slow. As @relayer464 reported, the execution of `php artisan icons:cache` fixed the slowness problem.

Man, god bless you, you save my life.

**Reply by argolian** [reply]
Worked for me too, thanks!

**Reply by Wsmallnews** [reply]
thanks, my problem has also been solved, why is the blade icon loading so slowly

**Comment by christoferd** [comment]
Hi Dan, 
These are the only Plugins I can think of - but they don't affect the Filament pages from what I can see:
![image](https://user-images.githubusercontent.com/1835883/230216574-bd03234c-934a-47a3-ba53-8971c8ab97d4.png)

As I'm now running through Apache, I don't have npm watching files. I actually don't have anything watching files.

Packages
See composer.json screenshots in the original post. 
It's been updated so they should be the latest stable versions. (phpStorm has displayed the current installed version numbers)

To add a **Tip for other Windows users**... make sure to exclude your development folders from Windows Defender (and any other Anti-Virus / anti-Malware, etc.)
- I did this today, but it didn't speed up Filament pages enough to merit any updates to page load times

I can see why my PC not running FastCGI is slower than the staging server, but I cannot work out why a blank Filament page would take over 1 second to load, while another admin page in the same app takes under a second.

**Comment by christoferd** [comment, marked as answer]
##Solved

**Solution: disable zend_extension Xdebug in php.ini**

Using the Performance Monitor I could track what was reading/writing the most data to my drive.
"cachegrind" ... which I found out is used by the Zend Extension Xdebug! 

Okay, but I can't be the only one using X-Debug. Do other people only enable it when it's absolutely necessary?

PS: Filament pages still take 2x longer to load, though now at 500ms, so that's much better. 
Or the page with 3 columns of all relational data, 10 records, takes 630ms

**Reply by fikurimax** [reply]
I have the same problem, every page or click is taking too much time like 5 - 10 seconds. But I dont have Xdebug installed

**Reply by Saifallak** [reply]
@fikurimax same over here, using nigix 1.24

**Reply by fikurimax** [reply]
@Saifallak  try to run ```php artisan icons:cache```

**Reply by Saifallak** [reply]
yes, I have done that, with no luck, requests taking 5+ seconds

**Reply by fikurimax** [reply]
if you're using debugbar, disable the view collector in the config file. Or try to optimize / cache config, view, etc.

**Reply by Saifallak** [reply]
nope not using it. but when i try, i see it took 5+secs in filament controller

**Reply by relayer464** [reply]
Most of my page requests take around 5 secs to load as well.  Xdebug is disabled in my php config.  Just wondering if @Saifallak or anyone else found a solution other than Xdebug?

**Reply by danharrin** [reply]
Do you have debugbar?

**Reply by Saifallak** [reply]
@danharrin nope,
Is there setting for nigix to optimize it?

**Reply by danharrin** [reply]
Opcache optimises it

**Reply by relayer464** [reply]
Thank you for the responses and suggestions.  I had read the debugbar can slow things down.  However, originally I did not have debugbar installed and only installed it later to help diagnose the problem.  I have since removed debugbar and it still takes about 5 sec for most pages to load.  

I am not currently using Opcache and was about to install it but then read that it is mostly useful in high demand production situations which isn't applicable to my situation.   

I am not sure if this is related but I am currently developing locally on my Windows 11 PC using Laragon.   I am just using the default nigix settings.  My frontend pages for the same app work fine (i.e., no significant delay).  Are there recommended nigix settings that would help to optimize it?

Any ideas or suggestions are much appreciated.  Thanks!

**Reply by danharrin** [reply]
Try `php artisan blade-icons:cache`

**Reply by relayer464** [reply]
Thanks for the suggestion.   I was able to cache the icons using the `php artisan icons:cache`  command (`php artisan blade-icons:cache`  didn't work), but unfortunately the pages are still slow.

I noticed there were slightly newer versions of nginx and PHP available, so updated both but still no change.

**Reply by Saifallak** [reply]
having the same problem with Nigix only, with apache everything running faster, maybe it's livewire a related problem?

**Reply by carlosmoran092** [reply]
I spent several hours looking for a way to optimize my application, it turns out that everything was solved with the `php artisan icons:cache `command. it went from taking 6 seconds to 142ms

**Reply by Saifallak** [reply]
Opcache worked for me,

**Reply by sloan58** [reply]
`php artisan icons:cache` FTW!

**Reply by pathros** [reply]
Indeed! I have been struggling working with Filament PHP both in local and production environment. When I started a new project in Filament PHP v3, the application started extremely fast, especially because I thought it was also thanks to the new version of Livewire. I also use lots of blade ui-kit-icons. So, when I started to install several packages of those blade icons, then I noticed that my Filament v3 turned slow again! ... so when I read the answer of @relayer464 `php artisan icons:cache` it kinda made sense to me. So I tried it now in local environment and it got fast again!

Now back to the older Filament PHP application, I executed the `php artisan icons:cache` command in local mode, and the application got fast as well. In production mode, it got quite a bit faster.

**Reply by victorycodedev** [reply]
Hello, i ran this command on my filament app in production, but my site is still extremely slow.  I use Cpanel. do you have any ideas on what else i can do?

**Reply by danharrin** [reply]
Shared hosting is notoriously slow, a cheap $5 VPS from DigitalOcean will always perform better

**Reply by victorycodedev** [reply]
> Shared hosting is notoriously slow, a cheap $5 VPS from DigitalOcean will always perform better

Yes you're right, cause I have another filament app on digital ocean working much better. I'll try to convince my client to consider moving there too. Thanks for your reply ❤️

**Reply by vdBergMike** [reply]
@sloan58 Thanks a lot for this one!

I created a website, was blazingly fast. Changed from Nova to Filament; BOOM! All pages in the website, loading for around 10 seconds. This solved the slow loading for all website pages.

**Reply by mlategan** [reply]
**php artisan icons:cache**
Amen to this!  10s down to few milliseconds.

**Reply by rahmap** [reply]
Thanks Bro

**Reply by vortechron** [reply]
@mlategan thanks, its work, hoping filamentphp has some sort of mechanism that show prompt to cache this icons on dev mode, putting on docs is easy to miss and hard to debug.

**Reply by bilogic** [reply]
anyone knows why `php artisan icons:cache` helped?

**Reply by mlategan** [reply]
> anyone knows why `php artisan icons:cache` helped?

In Blade Icons' Factory.php file, the registerComponents function attempts to load all icons needed for component registration from a Manifest class by calling its getManifest method. The Manifest class first checks its cache array, called manifest for the value requested. If the icon is present there, it returns it immediately, avoiding any further scanning. However, if the icon isn’t found in the cache, the build method is triggered, which iterates through all icon sets registered with the package. This method traverses each registered path in each set, scanning through all files in the paths and then building the manifest. 

On subsequent calls within the same request, icons will then resolve directly from this manifest.

However, with Livewire components (such as Filament forms, fields, etc.), multiple calls are made to the server to dynamically update parts of the page, resulting in multiple requests. Consequently, the manifest may be rebuilt for each request, leading to performance slowdowns.

This happens if component registration is switched on in Blade Icons, because the Provider for the package must register **all** these components with blade before blade can resolve them. If only the @svg function is used and component registration is disabled, it simply resolves each icon at the point in time at which it is rendered.

Running `php artisan icons:cache` pre-generates the manifest and saves it to a file in the application’s cache directory. This approach ensures that, instead of rebuilding the manifest repeatedly, Blade Icons can simply read from the cached manifest file, significantly improving performance.

Blade Icons and recently also Filament's documentation both inform the user of this performance issue.

Refer to:
https://github.com/blade-ui-kit/blade-icons?tab=readme-ov-file#caching
https://filamentphp.com/docs/3.x/panels/installation#caching-blade-icons
https://filamentphp.com/docs/3.x/panels/upgrade-guide#blade-icon-components-have-been-disabled

And the sources:
https://github.com/blade-ui-kit/blade-icons/blob/1.x/src/Factory.php
https://github.com/blade-ui-kit/blade-icons/blob/1.x/src/IconsManifest.php

**Comment by bhdbhd** [comment]
I am getting desperate. I have the same issue. Tried all mentioned methods: disabling xdebug and caching icons, also disabled observing views in debugbar. 

Still on my local environment initial response takes 9-11 seconds and loading scripts takes 2 seconds each. My queries are trivial and take less than 3ms. 

![filament-capture](https://github.com/filamentphp/filament/assets/5315278/4ae1b24f-4b25-4dea-a5a8-878fe59a84eb)
![filament-queries](https://github.com/filamentphp/filament/assets/5315278/5468c948-6a12-4b38-9687-8d45faf74fc1)

**Reply by sergiuprt** [reply]
I'm actually in the same situation. Don't know what to try anymore...

**Reply by bhdbhd** [reply]
Try turning opcache on in php.ini. I was banging my head against the wall but this seems to work with all the rest of suggestions

**Reply by christoferd** [reply]
@bhdbhd @sergiuprt Are you using Windows? Did you add your web project's root folder to Windows Defender Exclusions list?

* Make sure Windows Defender is not scanning every single file when accessing
* **Add the project's root web folder to Windows Defender Exclusions list**

**Reply by D3m0nslay7** [reply]
can say, all these 3 plus disabling xdebug made it load in ms now. compared to 5 - 10 seconds before

**Comment by NikunjGupta01** [comment]
> **What have I done wrong? Filament pages running extremely slow.** Note that all the load times examples are 2nd attempts with server cache (apcu) and browser cache enabled (unless the image shows cache disabled).
> 
> #### TLDR; Update 5-Apr-2023
> Filament Dashboard (Debug & Debug Bar = true): Local PC & Staging servers: ~1.6s
> 
> Other dashboard (no Filament) Local PC & Staging servers: ~0.8s
> 
> * Filament page takes 2x longer to load with just the base layout view including navigation. Why?
> 
> **Noting differences between Local PC and Remote Server**
> 
> * Remote runs FPM/FastCGI - local PC doesn't
> * Remote runs Thread-Safe disabled - local PC runs Thread Safe
> * Remote server CPU is probably much faster
> * Remote server is not Windows
> 
> **Filament Table/List:** 10 records per page 3 columns, each cell is related information from another table. Local PC (Debug & Debug Bar On):
> 
> * 174 models!
> * 41MB!
> * 11.32s!
>   ![image](https://user-images.githubusercontent.com/1835883/230112563-57bf4ffc-0df1-4cbd-b4e3-4b475f6f3549.png)
> 
> Local PC (Debug & Debug Bar Off):
> 
> * 7s!
> 
> Staging Server (Debug & Debug Bar Off):
> 
> * 1.3s
>   **Why is the remote server running faster? What tools are there to troubleshoot this?**
>   Note that changing APP_ENV=local to something else, has no effect on the load times.
> 
> Everything is setup following instructions from the Filament documentation site. I'm new to this, so I've only been following the instructions.
> 
> I would like to set this to 50 records per page, but then it will reach the 40+seconds load time 👎 I'm guessing there's a better way to do this, by overriding the query used to grab the records, but it goes against the principle reason for using this Filament tool/framework. And then if I did try do it, I can see issues when I want to use ordering, filters etc.
> 
> ## Staging Server 5-Apr-2023
> This runs similar to local PC. Weird though, because debug bar is off, so I'd expect it to go much faster! APP_DEBUG=true DEBUGBAR_ENABLED=false
> 
> Filament Dashboard - blank, without Widgets ![image](https://user-images.githubusercontent.com/1835883/230108966-f1086930-3580-4dd4-891d-701cd1b012ba.png)
> 
> My dashboard - no Filament
> 
> ![image](https://user-images.githubusercontent.com/1835883/230109261-57186e95-7436-410f-bc97-626692519709.png)
> 
> ## Local Windows 11
> This is a new app, Laravel v10.5.1 PHP 8.1.17 Windows 11 Server using CLI: php artisan serve running on 127.0.0.1:8000 Laravel cache "apc" No memcached No redis Database local Maria 10.x
> 
> I have reached 1590 views in a single request! 48MB! So looking at other discussions, this may be something to do with DebugBar?... but Filament is definitely doing something to slow down the app.
> 
> ![image](https://user-images.githubusercontent.com/1835883/229908912-bb114c82-d237-49e4-aab2-b7b633339de1.png)
> 
> Notice that it took 42s to load. That's on my development notebook. Recently purchased, running M.2 drive. All other apps run very quick.
> 
> As I add more resources, the loading time increases. I don't understand why.
> 
> I can post some pages of content, but I can't just dump all my code for public view.
> 
> With APP_DEBUG=false & DEBUGBAR_ENABLED=false...
> 
> ![image](https://user-images.githubusercontent.com/1835883/229910073-2e90d7fc-8974-45f4-8df5-96612cf83324.png)
> 
> This is one of my own routes that doesn't use filament. Notice the 390ms initial file load... now compare that to the previous one!
> 
> ![image](https://user-images.githubusercontent.com/1835883/229910412-05f77913-84d4-482c-a66d-e25cdeb3670e.png)
> 
> ## Composer
> ![image](https://user-images.githubusercontent.com/1835883/229917008-e5e00575-37d4-451b-8914-c73d5c8b2a2a.png) ![image](https://user-images.githubusercontent.com/1835883/229917116-e4b75684-8742-4372-bf9f-36d26a738e98.png)
> 
> ## More troubleshooting...
> No Debug, after setting google_fonts to NULL, it still takes the same amount of time, 10s to load first html.
> 
> ![image](https://user-images.githubusercontent.com/1835883/229913155-58f3bac7-ffe1-452b-a5b6-f63f1765de37.png)
> 
> I enabled OpCache, it now loads first html in 9.3 seconds :(
> 
> ## Switched to Apache 2.4
> Using XAMPP instead now, same PHP v8.1.X APP_DEBUG=false & DEBUGBAR_ENABLED=false It takes 1.2 seconds to load initial html. That is still very slow compare to the other page within the same app. ![image](https://user-images.githubusercontent.com/1835883/229918208-c467947b-f74a-43f0-bd26-28790f2a6f4d.png)
> 
> ## Debug back on
> APP_DEBUG=true & DEBUGBAR_ENABLED=true 5.89s ![image](https://user-images.githubusercontent.com/1835883/229923780-738cbbe4-742b-4ba1-a8fc-678ea578f8b7.png)
> 
> vs my own dashboard route that doesn't use Filament... 0.69s ![image](https://user-images.githubusercontent.com/1835883/229924023-6164d199-31fe-4614-b29a-fd8b24914294.png)
> 
> **Queries** What happens if I keep going and add another 20 resources?
> 
> Why is Filament acting so differently with artisan serve (PHP CLI server) vs Apache?

its just off topic so sorry already but how did u got filament projects running on xampp ? Because anyting i try just gets me to login page but it just loops there.

**Comment by catative** [comment]
I also encountered this problem. On a completely clean installation of Laravel 10.40.0 and Filament 3.1.42, I created one simple model, one migration for it with one custom text field and one Filament resource. Going (in UI) to the section of the created resource or using any action, for example, creating a new model record in DB, took a very long time (about 10 seconds). Disabling the Laravel debug panel (barryvdh/laravel-debugbar) or xDebug did not result in success. The problem was on both the Apache and Nginx servers (on Ubuntu 22.04).

Experiencedly, I noticed that this problem began to occur with Filament 3.0.28 version (in 3.0.27 everything worked quickly). It turned out that in my case the problem was due to the **Adblock Plus extension** for the Google Chrome browser. When I turned it off, everything immediately began to work quickly even in the latest versions of Filament and even with enabled xDebug.

**Reply by Sjoerd24** [reply]
Thank you so much! This also solved my problem! So strange that adblocker is interfering so much..

**Reply by danharrin** [reply]
Its a recent change, Adblock used to be fine. Now, its fighting with normal JavaScript in Chrome.

**Reply by alsahlawi** [reply]
Wow. disable adblock solved my issue, many thanks!!!

**Comment by geowrgetudor** [comment]
Maybe related, but not sure. I've came across a similar issue, but on the frontend side on LOCAL env (even with a similar setup as production - caching, etc.). 

It turned out **the issue was barryvdh's debugbar** which was causing UI to freeze sometimes even for 6-7s. It's probably of the high amount of views Filament uses because it tries to trace them all.

**Reply by scottzirkel** [reply]
You can disable the views from the debugbar in `config/debugbar.php`. set `collectors['views']` to false.

**Comment by ivl04** [comment]
I've implemented several optimizations, including caching icons, disabling the debugger bar, removing unnecessary packages, and performing a fresh Composer installation. Additionally, I disabled the Zend extension in the 'php.ini' file. Despite these efforts, the performance issue persists. Specifically, the /livewire/update is being called twice initially on every page/route. Even for simple view pages with only one record, it takes more than 15 seconds to load. On listing pages with only 10 paginated records, it takes around 20 seconds to load, with some JavaScript taking additional time after the page is fully rendered. Any advice on how to address these performance issues?

**Comment by sv-hmelevsky** [comment]
The problem has not been solved and still exists. Clean installation of `PHP8.2, Laravel 11, Filament 3.2, laravel-debugbar`. OPcache is enabled. Xdebug not installed. Disabled all extensions in the browser, ran: `php artisan icons:cache && php artisan filament:cache-components && php artisan optimize`. The cold basic load is 11-12sec, subsequent 3-8sec. And this is on a top-end PC on Local with an empty database and zero load! It’s scary to imagine what will happen in production! What to do? Any advice?

**Reply by Darnes1958** [reply]
> The problem has not been solved and still exists. Clean installation of `PHP8.2, Laravel 11, Filament 3.2, laravel-debugbar`. OPcache is enabled. Xdebug not installed. Disabled all extensions in the browser, ran: `php artisan icons:cache && php artisan filament:cache-components && php artisan optimize`. The cold basic load is 11-12sec, subsequent 3-8sec. And this is on a top-end PC on Local with an empty database and zero load! It’s scary to imagine what will happen in production! What to do? Any advice?

yes, The problem has not been solved and still exists . I do every thing wirtten in web but it is slow even with empty data in local.
filament is the best only on slow

**Reply by danharrin** [reply]
Personally I have never experienced numbers like this. If it was a more widespread issue we would have looked into it by now, but its not something we can reproduce easily and it seems to affect only a few people (I know there are many people in this thread, but relatively we have thousands of Filament applications).

Are you willing to create a new repository with a basic slow Filament app and share it with us so we can try to reproduce the problem again?

**Reply by Darnes1958** [reply]
> Personally I have never experienced numbers like this. If it was a more widespread issue we would have looked into it by now, but its not something we can reproduce easily and it seems to affect only a few people (I know there are many people in this thread, but relatively we have thousands of Filament applications).
> 
> Are you willing to create a new repository with a basic slow Filament app and share it with us so we can try to reproduce the problem again?

Slowness usually occurs on first load of system or page from menu , when press action button or the action button take much time to be active , when  press chekbox or toggle which have dependant with other component like toggle  (any operation use Get and Set) and more .
Plz Mr danharrin, solve the slowness is all filment need  to become the best .

**Reply by danharrin** [reply]
Without being able to reproduce the problem. I can't track down what causes it and fix the underlying issue.

**Reply by sv-hmelevsky** [reply]
> Without being able to reproduce the problem. I can't track down what causes it and fix the underlying issue.

Thank you very much for your reply! I uploaded my project for tests: 
https://github.com/sv-hmelevsky/filament-test-slow-app/tree/main 

Any help would be much appreciated! 🙏

**Reply by sv-hmelevsky** [reply]
> Without being able to reproduce the problem. I can't track down what causes it and fix the underlying issue.

Good afternoon! Did you manage to deploy/test the project?

**Reply by danharrin** [reply]
125-160ms to load the dashboard, using your Docker image

<img width="1624" alt="Screenshot 2024-07-01 at 14 18 08" src="https://github.com/filamentphp/filament/assets/41773797/dd531092-6cd2-4db1-a2b1-749c7de0f53b">

**Reply by sv-hmelevsky** [reply]
> 125-160ms to load the dashboard, using your Docker image
>
Thank you for your reply! I don't understand why this is happening! I have 5.5-6.3s.  What could be such a big difference? What OS are you using by the way? Do you have any guesses as to why this might be happening on Win 10 + WSL 2?

![image](https://github.com/filamentphp/filament/assets/4187388/2a894032-3fbf-4934-8338-741de11a05b7)

**Reply by danharrin** [reply]
What does the "timeline" tab say in debugbar

**Reply by sv-hmelevsky** [reply]
> What does the "timeline" tab say in debugbar
![image](https://github.com/filamentphp/filament/assets/4187388/f7db6689-09aa-48c8-8eaf-de309a644892)

**Reply by krekas** [reply]
With wsl it can be slow because of using wrong file system if I remember correctly. You have to make a research about this case.

**Reply by danharrin** [reply]
I don't use Windows, I thought WSL was an alternative to using Docker? In this case the reproduction repository uses Docker. If anyone with a Windows machine could pull it down and see if they can reproduce the slowness that would be good.

**Reply by sv-hmelevsky** [reply]
> With wsl it can be slow because of using wrong file system if I remember correctly. You have to make a research about this case.
>
@krekas That's very strange. Is there any way to configure WSL2 to avoid such problems? 
And as @danharrin said it would be cool if someone on Windows 10 + WSL tried to reproduce the problem.

**Reply by krekas** [reply]
As I said do a research. There is 100% others with such problem. This has nothing to do with filament or even laravel.

**Reply by ctoma** [reply]
@sv-hmelevsky I am guessing you have your project files under the windows mount instead of under WSL2. That's probably the cause of the slowness and is what krekas is referring to. 

Move all your project files into WSL2 and access them from phpstorm/vscode using WSL. 

For reference I am using WSL2 + Docker and load my empty Filament dashboard in 50ms. Other pages depend a lot on the # of rows and columns, but for me 10 rows & 10 columns load in about 200ms.

**Reply by christoferd** [reply]
FYI - This was marked as [Solved] as I was not using Docker. 
I avoid using Docker due to all the reasons above, I just cannot find a way to make it run as fast as I like on Windows. Then having to map a folder to edit the files etc. It's extra overhead I don't need at the moment and would only manage a project like that if it is 100% necessary.
I am however using Windows 11, and "php artisan serve" as it's much easier and fast enough.

**Reply by artulance** [reply]
Hi, I was recently experiencing the same problems as described in this thread. I am using windows 10, php artisan serve and php-8.2.4-Win32-vs16-x64.
I tried everything when caching laravel, icons etc but it was still taking too long sometimes 15 seconds.
The solution was to add the project folder to the windows defender exclusion, now it only takes milliseconds, it's incredible.

**Reply by Darnes1958** [reply]
> Hi, I was recently experiencing the same problems as described in this thread. I am using windows 10, php artisan serve and php-8.2.4-Win32-vs16-x64. I tried everything when caching laravel, icons etc but it was still taking too long sometimes 15 seconds. The solution was to add the project folder to the windows defender exclusion, now it only takes milliseconds, it's incredible.

Yes I add the folder to "windows defender exclusion" , I do it in the local and in the server and the two is good speed , thank you .
And also when run "filament:optimize" in the product and do OPcache it change to better with me.

**Comment by hamza-siddique1** [comment]
I blocked the font CDN request (in chrome dev tools) which was taking too long to load
![image](https://github.com/user-attachments/assets/0a34299e-3101-4d70-aa64-a76c9850fa64)

Now it is super fast.

**Reply by Darnes1958** [reply]
> I blocked the font CDN request (in chrome dev tools) which was taking too long to load ![image](https://private-user-images.githubusercontent.com/57829074/361124779-0a34299e-3101-4d70-aa64-a76c9850fa64.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxODc1MTAsIm5iZiI6MTcyNjE4NzIxMCwicGF0aCI6Ii81NzgyOTA3NC8zNjExMjQ3NzktMGEzNDI5OWUtMzEwMS00ZDcwLWFhNjQtYTc2Yzk4NTBmYTY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEzVDAwMjY1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4ZDQ0NzdlOTcyZmVkOGQwNzc3ODU4YWY1ZjhhYzA2Mzg3YTlkYTMzYjE5ODRiZmVjOGI3Zjc0OGI3YzcxMDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.mTxeXK5qzJ0OOEDMaCkoeMidScdwcba0F7h7zEzNEBw)
> 
> Now it is super fast.

how to do that

**Reply by hamza-siddique1** [reply]
Here I described the method in details 
https://stackoverflow.com/a/78908426/25138504

**Comment by christoferd** [comment]
Windows developers should note one of the tips here
"The solution was to **add the project folder to the windows defender exclusion**, now it only takes milliseconds, it's incredible." - artulance

**Reply by emmanpbarrameda** [reply]
UP FOR THIS!

**Comment by sv-hmelevsky** [comment]
> @ctoma Move all your project files into WSL2 and access them from phpstorm/vscode using WSL.

I did a little "research" in this area, you were right! The problem was in the file system between Windows and WSL. I moved the files to WSL, now Filament loads very quickly! Here is a small "guide".

I will say right away that I work with WSL as root, so I created it as root. The practice is not very good, but you can always do this for a separate user, replacing the directories with your own.

1. Created a directory for **projects** in WSL: `mkdir -p /root/projects`
2. Installed **inotify-tools** and **rsync** in WSL. This will help us synchronize our working folder with projects on Windows with the project folder on WSL. 
3. Created a script file for automatic synchronization between Windows and WSL: `mkdir -p /root/scripts && nano /root/scripts/sync_projects.sh`

```
#!/bin/bash

# Path to projects in the Windows file system
WINDOWS_PROJECTS_PATH="/mnt/e/server/domains/"
# Path to projects in WSL
WSL_PROJECTS_PATH="/root/projects/"

# Launch inotifywait to track changes
inotifywait -m -r -e modify,create,delete $WINDOWS_PROJECTS_PATH | while read path _ file; do
    # Synchronizing changes via rsync
    rsync -av --delete $WINDOWS_PROJECTS_PATH $WSL_PROJECTS_PATH
done
```
5. Made the script executable: `chmod +x /root/scripts/sync_projects.sh` and ran it. Waited for the synchronization to finish (it could take some time).
6. Added background script launch: `echo "nohup /root/scripts/sync_projects.sh &" >> /root/.bashrc`
7. Added additional (local) docker-compose config to the project: **docker-compose.override.yml** and redefined the paths in it, from this: `- ./client-service/app:/var/www/client-service` to this: `- ~/projects/atb-client/client-service/app:/var/www/client-service`
8. **Made a DB dump!**
9. Rebuilt docker: `docker-compose down && docker-compose build && docker-compose up -d`
10. That's it! Now any changes in the project files on Windows cause synchronization with WSL and everything works very quickly! In theory, you can not use synchronization, but work in the IDE directly with the WSL directory. But I don't really like this concept. :\ I hope this "guide" will be useful to someone. :)

**Comment by sasaorasanin** [comment]
One thing on this topic, I had same issue that none of above answers solved. Weirdly `->blocks(fn () => [elements])` was causing it but `->blocks([elements])` worked very fast. So maybe check where you use fallbacks, that may be an issue

**Comment by waggos-root** [comment]
Hi, I deployed my application on two servers and I can't explain why is taking 4-6 seconds to go from resource A to resource B only on one of them. I already tried the suggestions found on this page.

**Reply by bilogic** [reply]
some lengthy SQL query?

**Reply by waggos-root** [reply]
Not at all, the issue arises when I go from a resource to another. Once a resource is loaded, pagination is fast (around 0.5 ms or less).

Here are a few steps I made to improve response timings:
- Enabled MySQL slow query log, and added indexes where needed.
- Installed memcached and made sure Filament is using it.
- ran php artisan optimize 
- ran php artisan filament:optimize
- and then
- php artisan icon:cache 
- php artisan config:cache 
- php artisan route:cache 

The same code is running on another server, and that one is fast.

**Reply by bilogic** [reply]
sounds like you suspect it is the server, if the same happened to me, i would wipe my server, which is a docker container

**Comment by heruprambadi** [comment]
on .env

APP_ENV=production
APP_DEBUG=false

this is works on me
